File: src/main/java/me/aikovdp/biggiebot/BiggieBot.java
package me.aikovdp.biggiebot;

import me.aikovdp.biggiebot.modules.Starboard;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.JDABuilder;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import net.dv8tion.jda.api.interactions.commands.build.Commands;
import org.jetbrains.annotations.NotNull;

import javax.security.auth.login.LoginException;


public class BiggieBot extends ListenerAdapter {

    public static void main(String[] args) throws LoginException {
        JDA jda = JDABuilder.createDefault(args[0])
                .addEventListeners(new BiggieBot(), new Starboard())
                .build();

        jda.updateCommands()
                .addCommands(Commands.slash("ping", "Ping the bot"))
                .queue();
    }


    @Override
    public void onSlashCommandInteraction(@NotNull SlashCommandInteractionEvent event) {
        if (event.getName().equals("ping")) {
            event.reply("pong").queue();
        }
    }
}

File: src/main/java/me/aikovdp/biggiebot/BiggieBot.kt
package me.aikovdp.biggiebot

import dev.minn.jda.ktx.events.onCommand
import dev.minn.jda.ktx.events.onContext
import dev.minn.jda.ktx.interactions.commands.message
import dev.minn.jda.ktx.interactions.commands.slash
import dev.minn.jda.ktx.interactions.commands.updateCommands
import dev.minn.jda.ktx.jdabuilder.light
import me.aikovdp.biggiebot.modules.Starboard
import me.aikovdp.biggiebot.modules.getStarboardChannel
import me.aikovdp.biggiebot.modules.sendToStarboard
import net.dv8tion.jda.api.entities.Message

fun main() {
    val jda = light(System.getenv("DISCORD_BOT_TOKEN"), enableCoroutines = true) {
        addEventListeners(Starboard())
    }

    jda.updateCommands {
        slash("ping", "Ping the bot")
        message("Send to starboard")
    }.queue()

    jda.onCommand("ping") {
        it.reply("pong").queue()
    }

    jda.onContext<Message>("Send to starboard") {
        sendToStarboard(
            getStarboardChannel(it.guild) ?: return@onContext,
            it.target
        )
        it.reply("Sent to starboard!").setEphemeral(true).queue()
    }
}
